test/Constraints/function_conversion.swift

```diff
@@ -68,3 +68,15 @@ noEscapeParam = takesAny // expected-error {{converting non-escaping value to 'A
 func rdar_59773317(x: () -> Int) -> (() -> Int)? { // expected-note {{parameter 'x' is implicitly non-escaping}}
   return x // expected-error {{using non-escaping parameter 'x' in a context expecting an @escaping closure}}
 }
+
+// rdar://problem/59703585 - Wrong error message when signature of a C function type and Swift implementation mismatch
+func rdar_59703585() {
+  typealias Fn = @convention(c) (UnsafePointer<Int8>?, UnsafeMutableRawPointer?) -> Void
+
+  func swiftCallback(someString: UnsafePointer<Int8>, someObject: UnsafeMutableRawPointer?) {}
+
+  var cb: Fn? = nil
+
+  cb = swiftCallback
```

Comment:
I wonder if for `AssignExpr` the diagnostic could be `cannot assign value of type 'A' to 'B'` instead of a `cannot convert value...`? :)

Reply:
Ah true, I'll fix later tonight!

Reply:
Fixed!